community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -851,6 +856,10 @@ private void release()
             securityContext = null;
             transactionEvent = null;
             explicitIndexTransactionState = null;
+            if ( txState != null )
+            {
+                txState.release();
+            }
             txState = null;
```

Comment:
looks like this should be inside of if statement

Reply:
Indeed.